raiden/utils/signer.py

```diff
@@ -0,0 +1,87 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+from eth_keys import keys
+from eth_keys.exceptions import BadSignature
+from eth_utils import keccak, to_checksum_address
+
+from raiden.exceptions import InvalidSignature
+from raiden.utils.typing import Address, AddressHex
+
+
+def eth_sign_sha3(data: bytes) -> bytes:
+    """
+    eth_sign/recover compatible hasher
+    Prefixes data with "\x19Ethereum Signed Message:\n<len(data)>"
+    """
+    prefix = b'\x19Ethereum Signed Message:\n'
+    if not data.startswith(prefix):
+        data = prefix + b'%d%s' % (len(data), data)
+    return keccak(data)
+
+
+def recover(
+        data: bytes,
+        signature: bytes,
+        hasher: Callable[[bytes], bytes] = eth_sign_sha3,
+) -> Address:
+    """ eth_recover address from data hash and signature """
+    _hash = hasher(data)
+
+    if signature[-1] >= 27:  # support (0,1,27,28) v values
```

Comment:
This code here accepts every kind of v. It just subtracts 27 if >=27. I would rather you were explicit and check if v in (0, 1, 27, 28) and if not bail.

Reply:
That's because `keys.ecdsa_recovery` accepts exclusively `[0, 1]` `v`s, so anything different from [0, 1, 27, 28] will trigger `BadSignature` -> `InvalidSignature`. I may leave a comment making that clearer

Reply:
Thank you, adding that comment would be nice. And I guess this is also already tested [here](https://github.com/raiden-network/raiden/pull/3386/files#diff-8ac59f39032ed7abd2238b37d11dbe10R37)